src/Token.php

```diff
@@ -20,11 +20,20 @@ class Token
 
     /**
      * @param string $value
-     * @param string $type
+     * @param string|null $type
      */
-    public function __construct(string $value, string $type)
+    public function __construct(string $value, ?string $type = null)
     {
         $this->value = $value;
-        $this->type = $type;
+        $this->type = $this->getTokenType($type);
```

Comment:
this is logic which should only happen in the stripe token type IMO, let's discuss.